src/test/java/com/google/cloud/spanner/jdbc/PgNumericResultSetTest.java

```diff
@@ -0,0 +1,801 @@
+import com.google.common.io.ByteSource;
+import com.google.protobuf.ListValue;
+import com.google.spanner.v1.ResultSetMetadata;
+import com.google.spanner.v1.StructType;
+import com.google.spanner.v1.StructType.Field;
+import com.google.spanner.v1.TypeAnnotationCode;
+import com.google.spanner.v1.TypeCode;
+import io.grpc.Server;
+import io.grpc.netty.shaded.io.grpc.netty.NettyServerBuilder;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.InetSocketAddress;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.DriverManager;
+import java.sql.NClob;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class PgNumericResultSetTest {
+
+  private static final String PROJECT = "my-project";
+  private static final String INSTANCE = "my-instance";
+  private static final String DATABASE = "my-database";
+  private static final String COLUMN_NAME = "PgNumeric";
+  private static final ResultSetMetadata SCALAR_METADATA =
+      ResultSetMetadata.newBuilder()
+          .setRowType(
+              StructType.newBuilder()
+                  .addFields(
+                      Field.newBuilder()
+                          .setName(COLUMN_NAME)
+                          .setType(
+                              com.google.spanner.v1.Type.newBuilder()
+                                  .setCode(TypeCode.NUMERIC)
+                                  .setTypeAnnotation(TypeAnnotationCode.PG_NUMERIC))))
+          .build();
+  private static final ResultSetMetadata ARRAY_METADATA =
+      ResultSetMetadata.newBuilder()
+          .setRowType(
+              StructType.newBuilder()
+                  .addFields(
+                      Field.newBuilder()
+                          .setName(COLUMN_NAME)
+                          .setType(
+                              com.google.spanner.v1.Type.newBuilder()
+                                  .setCode(TypeCode.ARRAY)
+                                  .setArrayElementType(
+                                      com.google.spanner.v1.Type.newBuilder()
+                                          .setCode(TypeCode.NUMERIC)
+                                          .setTypeAnnotation(TypeAnnotationCode.PG_NUMERIC)))))
+          .build();
+  private static final String QUERY = "SELECT " + COLUMN_NAME + " FROM Table WHERE Id = 0";
+  private static final int MAX_PG_NUMERIC_SCALE = 131_072;
+  private static final int MAX_PG_NUMERIC_PRECISION = 16_383;
```

Comment:
`131_072` or `16_383` do not look like `int`, can you explain the above 2 lines.

Reply:
Java actually allows underscores in int literals for better readability: https://docs.oracle.com/javase/7/docs/technotes/guides/language/underscores-literals.html

Reply:
Thanks, that is helpful.